Enhancements to Android version

Tempus35 opened the pull request:
I have been working on adding and refining features in the android version of the plugin. 

Changes so far:
-A end datetime that uses the expire function of the google api
-Added frequency and last shown time that will allow you to control when a notification will show, ie every hour you are in the fence etc.
-Added Dwell and usable functions from the google api for dwell
-Change the intent flags for the notification so the app will be pulled to the front if it is not unloaded instead of reloading etc time
-Due to above change, added code to detect new intent and run notification clicked (Might need work)
-Added in option for notification responsiveness via google api
-Variety of other changes

All changes are to Android only side at the moment and have not been full tested, working on that. I also will look into making similar changes, if needed to iOS etc.

tsubik commented:
Hey, thanks for PR. I'm in the middle of rewriting android part a little bit to get rid of deprecations, but with some changes, I could merge this pull request.

Tempus35 commented:
I changed the files based on your input, the getExpiredTime might need work. I also moved the store for last shown up, is that where you meant when you commented on it?

Tempus35 commented:
Had a quick question about iOS version, no clue where to put it and could not find online. Is there a similar function to the autoCancel function on the iOS version? IE Can I stop the notification from disappearing when clicked on in the iOS version.

Tempus35 commented:
Also realized for accuracy it might be better to send in a timestamp on endDate not a formatted string. Think I might change that

tsubik commented:
I have no idea of iOS has something like `autocancel`. If timestamp you should change name to `endAt` or maybe `expireAt`.

Tempus35 commented:
So something I changed or something from before is causing the receiveTransition service to crash after a while. I will be looking into this soon.
